optuna/integration/chainermn.py

```diff
@@ -49,7 +49,10 @@ def __call__(self, trial):
         # type: (Trial) -> float
 
         self.comm.mpi_comm.bcast((True, trial._trial_id))
-        return self.objective(trial, self.comm)
+        try:
+            return self.objective(trial, self.comm)
+        finally:
+            self.comm.mpi_comm.barrier()
```

Comment:
Without this barrier, there is a possibility that a delayed node calls (for example) `trial.suggest_xxx(...)` in the objective function even after the rank0 node has already changed the state of the trial to `COMPLETE`.